source/library/Signet/Unstable/Type/Test.hs

```diff
@@ -0,0 +1,20 @@
+{-# LANGUAGE RankNTypes #-}
+
+module Signet.Unstable.Type.Test where
+
+import qualified Control.Monad as Monad
+import qualified Data.Void as Void
+import qualified GHC.Stack as Stack
+
+data Test spec = MkTest
+  { assertFailure :: (Stack.HasCallStack) => String -> IO Void.Void,
+    describe :: String -> spec () -> spec (),
+    it :: String -> IO () -> spec ()
```

Comment:
Similarly, "it" could be "case", but that's annoying to work with since it's a keyword.